wgpu/src/lib.rs

```diff
@@ -3464,9 +3464,13 @@ impl Queue {
         Context::queue_write_buffer(&*self.context, &self.id, &buffer.id, offset, data)
     }
 
-    /// Schedule a data write into `buffer` starting at `offset` via the returned [QueueWriteBufferView].
+    /// Schedule a data write into `buffer` starting at `offset` via the returned
+    /// [QueueWriteBufferView].
     ///
-    /// The returned value can be dereferenced to a `&mut [u8]`; dereferencing it to a `&[u8]` panics!
+    /// The returned value can be dereferenced to a `&mut [u8]`; dereferencing it to a
+    /// `&[u8]` panics!
+    /// (It is not unsound to read through the `&mut [u8]` anyway, but doing so will not
+    /// yield the existing contents of `buffer`, and it is likely to be slow.)
```

Comment:
This is oddly phrased, it feels like you're not guaranteed to read the same value that you've just written, which would definitely be UB.

Reply:
I'm specifying that it _will not_ give the contents of `buffer`, the preexisting GPU buffer, and leaving unspecified what the initial value of this `[u8]` is. I agree this isn't the clearest wording; ideas welcome.

Reply:
I added a minor qualification to clarify this.